MonoGame.Framework.Content.Pipeline/Graphics/DxtTexture/BlockWindow.cs

```diff
@@ -0,0 +1,612 @@
+        /// 
+        /// </remarks>
+        /// <param name="rgba">The rgba values of the 16 source pixels.</param>        
+        /// <param name="flags">Compression flags.</param>
+        public void Compress(Byte[] rgba, CompressionMode flags)
+        {
+            // compress with full mask
+            CompressMasked(rgba, 0xffff, flags);
+        }
+        
+        /// <summary>
+        /// Compresses a 4x4 block of pixels.
+        /// </summary>
+        /// <remarks>
+        /// The source pixels should be presented as a contiguous array of 16 rgba
+        /// values, with each component as 1 byte each. In memory this should be:
+        /// 
+        ///     { r1, g1, b1, a1, .... , r16, g16, b16, a16 }
+        ///     
+        /// The mask parameter enables only certain pixels within the block. The lowest
+        /// bit enables the first pixel and so on up to the 16th bit. Bits beyond the
+        /// 16th bit are ignored. Pixels that are not enabled are allowed to take
+        /// arbitrary colours in the output block. An example of how this can be used
+        /// is in the CompressImage function to disable pixels outside the bounds of
+        /// the image when the width or height is not divisible by 4.
+        /// 
+        /// The flags parameter should specify either kDxt1, kDxt3 or kDxt5 compression, 
+        /// however, DXT1 will be used by default if none is specified. When using DXT1 
+        /// compression, 8 bytes of storage are required for the compressed DXT block. 
+        /// DXT3 and DXT5 compression require 16 bytes of storage per block.
+        /// 
+        /// The flags parameter can also specify a preferred colour compressor and 
+        /// colour error metric to use when fitting the RGB components of the data.
+        /// Possible colour compressors are: kColourClusterFit (the default), 
+        /// kColourRangeFit or kColourIterativeClusterFit. Possible colour error metrics 
+        /// are: kColourMetricPerceptual (the default) or kColourMetricUniform. If no 
+        /// flags are specified in any particular category then the default will be 
+        /// used. Unknown flags are ignored.
+        /// 
+        /// When using kColourClusterFit, an additional flag can be specified to
+        /// weight the colour of each pixel by its alpha value. For images that are
+        /// rendered using alpha blending, this can significantly increase the 
+        /// perceived quality.
+        /// </remarks>
+        /// <param name="rgba">The rgba values of the 16 source pixels.</param>
+        /// <param name="mask">The valid pixel mask.</param>
+        /// <param name="flags">Compression flags.</param>
+        public void CompressMasked(Byte[] rgba, int mask, CompressionMode flags)
+        {
+            System.Diagnostics.Debug.Assert(rgba != null && rgba.Length == 64, "rgba");
+
+            // fix any bad flags
+            flags = flags.FixFlags();

```

Comment:
This is really poor... for every block in a 2048x2048 (that would be 262,144 blocks) it fixes the flags? 

This should be done once per image compressed... not for each block.